File: webapp/src/domain/product.ts
export interface Seller {
  name: string;
  logoUrl: string;
  official: boolean;
  products: number;
  sales: string;
  indicators: { icon: string; text: string }[];
  storeLink: string;
}

export interface Features {
  icon: string;
  label: string;
  value: string;
  subValue?: string;
  hasBar?: boolean;
}

export interface Product {
  id: string;
  title: string;
  price: number;
  discount_percentage: number;
  installments: string;
  images: any[];
  color: string;
  stock: number;
  description: string;
  seller: Seller;
  payment_methods: any;
  features: Features[];
}

File: webapi/domain/models/product.py
from typing import List
from .seller import Seller
from .features import Features
from pydantic import BaseModel

class Product(BaseModel):
    id: str
    title: str
    price: float
    discount_percentage: int
    installments: str
    images: List[str]
    color: str
    stock: int
    description: str
    seller: Seller
    payment_methods: List[str]
    features: Features
